docs/rfcs/0004-rfc-connectiomanager.md

```diff
@@ -0,0 +1,80 @@
+- Status: In progress
+- Date: 2023-11-02
+- Author: [@gela](https://github.com/gela)
+- RFC PR: TBD
+- Implementation GitHub issue: TBD.
+
+
+## Summary
+
+The existing `ConnectionManager` component will handle all connectivity related checks once during VPN activation (`Controller::StateConnecting`) and then periodically once the VPN is already activated (`Controller::StateOn`); the controller will be the entity which manages the controller states that the client goes through such as `connecting` and `confirming` as we activate and deactivate the VPN to reflect those to the user what is happening behind the scene.
+
+## Motivation
+
+At present, the handling of VPN connectivity lacks consistency for users. There are instances when the VPN goes into No Signal due to a variety of connectivity-related issues, such as the network connection failing or the server location becoming unavailable. In these situations, users are often left without a clear explanation of what went wrong or guidance on how to troubleshoot and resolve the issue. They may encounter a background error message or lose signal without further information.
```

Comment:
> They may encounter a background error message or lose signal without further information.

Consider mentioning the actual error to clarify the motivation for this feature. For example, if the internet connection is disabled, a "No Signal" error is displayed, without indicating that it is caused by lack of an internet connection.

Also, what does "background" mean?

Reply:
> Also, what does "background" mean?

It's literally a "background error occurred" messaged with no additional context. I'm guessing it's referring to something going wrong with Guardian.